### PR TITLE
release/2.4.0 fixed OpenPepXL parallelization errors

### PR DESCRIPTION
**Pull request by enetz**
race conditions caused crashes, when LOG_DEBUG was written in from different threads.
Now all LOG_DEBUG writing operations are in critical sections and only compiled and run if they are activated using the `#define DEBUG_...`and `#ifdef DEBUG_...`conditions.

### CODE REVIEW

**Comment by enetz** [comment]
so essentially, most of the debug output is turned off now to avoid race conditions or many critical sections

File: src/openms/source/ANALYSIS/XLMS/OPXLHelper.cpp [hunk 1]
```diff
@@ -37,25 +37,23 @@
...
     // initialize empty vector for the results
     vector<OPXLDataStructs::XLPrecursor> mass_to_candidates;
 
     double min_precursor = spectrum_precursors[0];
     double max_precursor = spectrum_precursors[spectrum_precursors.size()-1];
 
-// Multithreading options: schedule: static, dynamic, guided
-// use OpenMP to run this for-loop on multiple CPU cores
-#ifdef _OPENMP
-#pragma omp parallel for schedule(static)
```
Comment (timosachsenberg) [inline src/openms/source/ANALYSIS/XLMS/OPXLHelper.cpp]: Why was that removed?

File: src/openms/source/ANALYSIS/XLMS/OPXLHelper.cpp [hunk 2]
```diff
@@ -195,16 +193,9 @@ namespace OpenMS
     if (low_it != up_it) // if they are not equal, there are matching precursors in the data
     {
       // found_matching_precursors = true;
-
-// don't access this vector from two processing threads at the same time
-#ifdef _OPENMP
-#pragma omp critical
```
Comment (timosachsenberg) [inline src/openms/source/ANALYSIS/XLMS/OPXLHelper.cpp]: Why not needed anymore

Comment (enetz) [reply src/openms/source/ANALYSIS/XLMS/OPXLHelper.cpp]: related to the removed nested parallel loop. Since the function itself is not parallel anymore and different calls to this function don't access the same instances of these vectors, there is no need for a critical section

File: src/openms/source/ANALYSIS/XLMS/OPXLHelper.cpp (hunk 1 above)
Comment (timosachsenberg) [reply src/openms/source/ANALYSIS/XLMS/OPXLHelper.cpp]: make sense

**Comment by timosachsenberg** [comment]
http://cdash.openms.de/viewBuildError.php?buildid=290498
